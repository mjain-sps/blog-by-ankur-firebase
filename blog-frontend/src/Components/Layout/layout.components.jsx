import React from "react";
import Header from "../Header/header.component";
import { LayoutContainer } from "./layout.styles.js";
const Layout = ({ children }) => {
  return (
    <>
      <Header />
      <LayoutContainer>{children}</LayoutContainer>
    </>
  );
};
export default Layout;
