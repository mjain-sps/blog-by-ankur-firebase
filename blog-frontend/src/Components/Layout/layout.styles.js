import styled from "styled-components";

const LayoutContainer = styled.div`
  margin-top: 60vh;
  @media (max-width: 576px) {
    margin-top: 42vh;
  }
`;

export { LayoutContainer };
