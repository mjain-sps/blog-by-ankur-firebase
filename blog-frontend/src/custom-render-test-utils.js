import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import rootReducer from "./Reducers";

const customRender = (
  ui,
  {
    initialState,
    store = createStore(rootReducer, initialState, applyMiddleware(thunk)),
    ...options
  } = {}
) => {
  const WrapperWithRoutes = ({ children }) => {
    return (
      <Provider store={store}>
        <BrowserRouter>{children}</BrowserRouter>
      </Provider>
    );
  };
  return render(ui, { wrapper: WrapperWithRoutes, ...options });
};

export * from "@testing-library/react";
export { customRender };
