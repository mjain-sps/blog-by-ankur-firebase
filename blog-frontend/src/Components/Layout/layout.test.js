import React from "react";
import { customRender } from "../../custom-render-test-utils";
import "@testing-library/jest-dom/extend-expect";
import Layout from "./layout.components";

test("Check snapshot layout", () => {
  const { container } = customRender(<Layout />);
  expect(container).toMatchSnapshot();
});
