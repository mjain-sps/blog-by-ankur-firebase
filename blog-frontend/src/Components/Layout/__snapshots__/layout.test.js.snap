// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check snapshot layout 1`] = `
<div>
  <div
    class="sc-bdnxRM fVWigx"
  >
    <div
      class="sc-gtsrHT bosWpy"
    >
      <div>
        Wed Jun 16 2021
      </div>
      <div
        class="sc-dlnjwi hkcgzi"
      >
        <div>
          <span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-facebook-square fa-w-14 sc-jrsJWt foguBv"
              data-icon="facebook-square"
              data-prefix="fab"
              data-testid="icon"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M400 32H48A48 48 0 0 0 0 80v352a48 48 0 0 0 48 48h137.25V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.27c-30.81 0-40.42 19.12-40.42 38.73V256h68.78l-11 71.69h-57.78V480H400a48 48 0 0 0 48-48V80a48 48 0 0 0-48-48z"
                fill="currentColor"
              />
            </svg>
          </span>
          <span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-linkedin fa-w-14 sc-jrsJWt foguBv"
              data-icon="linkedin"
              data-prefix="fab"
              data-testid="icon"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
                fill="currentColor"
              />
            </svg>
          </span>
          <span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-youtube fa-w-18 sc-jrsJWt foguBv"
              data-icon="youtube"
              data-prefix="fab"
              data-testid="icon"
              focusable="false"
              role="img"
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M549.655 124.083c-6.281-23.65-24.787-42.276-48.284-48.597C458.781 64 288 64 288 64S117.22 64 74.629 75.486c-23.497 6.322-42.003 24.947-48.284 48.597-11.412 42.867-11.412 132.305-11.412 132.305s0 89.438 11.412 132.305c6.281 23.65 24.787 41.5 48.284 47.821C117.22 448 288 448 288 448s170.78 0 213.371-11.486c23.497-6.321 42.003-24.171 48.284-47.821 11.412-42.867 11.412-132.305 11.412-132.305s0-89.438-11.412-132.305zm-317.51 213.508V175.185l142.739 81.205-142.739 81.201z"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
        <div />
        <div>
          <a
            data-testid="signin-link"
            href="/signin"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-sign-in-alt fa-w-16 "
              data-icon="sign-in-alt"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 448h-84c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h84c17.7 0 32-14.3 32-32V160c0-17.7-14.3-32-32-32h-84c-6.6 0-12-5.4-12-12V76c0-6.6 5.4-12 12-12h84c53 0 96 43 96 96v192c0 53-43 96-96 96zm-47-201L201 79c-15-15-41-4.5-41 17v96H24c-13.3 0-24 10.7-24 24v96c0 13.3 10.7 24 24 24h136v96c0 21.5 26 32 41 17l168-168c9.3-9.4 9.3-24.6 0-34z"
                fill="currentColor"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
    <div
      class="sc-hKFxyN KshaT"
    >
      <img
        class="sc-eCApnc hhsRHD"
        src="Logo.jpeg"
      />
    </div>
    <div
      class="sc-jSFjdj gqeyFR"
    >
      <div
        class="sc-gKAaRy gUbSYU"
      >
        <a
          class="sc-fujyAs kBdeKt"
          currentpath="CURRENT"
          href="/"
          name="home"
        >
          HOME
        </a>
        0
      </div>
      <div
        class="sc-pNWdM fbzoKR"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-search fa-w-16 sc-jrsJWt foguBv"
          data-icon="search"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    class="sc-iemWCZ kzoNGX"
  />
</div>
`;
